### PR TITLE
Remove unused diagnostic methods.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs [hunk 1]
```diff
@@ -108,49 +108,8 @@ public bool IsDiagnosticSuppressed(Diagnostic diagnostic, out AttributeData supp
             return false;
         }
 
-        private bool IsDiagnosticSuppressed(Diagnostic diagnostic, out SuppressMessageInfo info, ISymbol symbolOpt = null)
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs]: there are no callers of this method that pass in "symbolOpt".  So symbolOpt is always null. So nothing ever called `IsDiagnosticSuppressed(diagnostic.Id, symbolOpt, out info)`.

File: src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs [hunk 2]
```diff
@@ -108,49 +108,8 @@ public bool IsDiagnosticSuppressed(Diagnostic diagnostic, out AttributeData supp
...
-            {
-                return true;
-            }
-
-            // Check for suppression on parent symbol
-            var parent = symbol.ContainingSymbol;
-            return parent != null && IsDiagnosticSuppressed(id, parent, out info);
-        }
+        private bool IsDiagnosticSuppressed(Diagnostic diagnostic, out SuppressMessageInfo info)
+            => IsDiagnosticSuppressed(diagnostic.Id, diagnostic.Location, out info);
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs]: this is just the original method, with the optional symbol (that was never used) removed.

**Comment by CyrusNajmabadi** [comment]
Tagging @jcouv

**Comment by jcouv** [comment]
test windows_debug_spanish_unit32_prtest  please

**Comment by jcouv** [comment]
Thanks!

**Pull request merged**